### DISCUSSION TITLE
[AssetMapper] Could not find an asset mapper path

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#51918] [Q&A] [answered]
I'm removing webpack from a fairly simple application and stuck on an AssetMapper issue.  The repo is at https://github.com/tacman/stenope-skeleton

```bash
git clone git@github.com:tacman/stenope-skeleton.git steno && cd steno && composer install && bin/console debug:asset-map
```

In ControllersMapGenerator.php line 127:
                                                                                                                                       
     Could not find an asset mapper path that points to the "swup" controller in package "symfony/ux-swup", defined in controllers.json.  
                                                                                                                                       
My setup seems so straight-forward, I've installed symfony/stimulus-controller and symfony/ux-swap and let the recipes work.

## controllers.json
```json
{
    "controllers": {
        "@symfony/ux-swup": {
            "swup": {
                "enabled": true,
                "fetch": "eager"
            }
        }
    },
    "entrypoints": []
}
```

## importmap.php
```php
<?php
return [
    'app' => [
        'path' => 'app.js',
        'preload' => true,
    ],
    '@hotwired/stimulus' => [
        'url' => 'https://cdn.jsdelivr.net/npm/@hotwired/stimulus@3.2.2/+esm',
    ],
    '@symfony/stimulus-bundle' => [
        'path' => '@symfony/stimulus-bundle/loader.js',
    ],
    '@swup/fade-theme' => [
        'url' => 'https://cdn.jsdelivr.net/npm/@swup/fade-theme@1.0.5/+esm',
    ],
    '@swup/slide-theme' => [
        'url' => 'https://cdn.jsdelivr.net/npm/@swup/slide-theme@1.0.5/+esm',
    ],
    '@swup/forms-plugin' => [
        'url' => 'https://cdn.jsdelivr.net/npm/@swup/forms-plugin@2.0.1/+esm',
    ],
    '@swup/debug-plugin' => [
        'url' => 'https://cdn.jsdelivr.net/npm/@swup/debug-plugin@3.0.0/+esm',
    ],
    'swup' => [
        'url' => 'https://cdn.jsdelivr.net/npm/swup@3.1.1/+esm',
    ],
];
```

Everything is current (Symfony 6.3.5)
```shell
composer outdated
Info from https://repo.packagist.org: #StandWithUkraine

Direct dependencies required in composer.json:
Everything up to date

Transitive dependencies not required in composer.json:
Everything up to date
tac@tac-desktop1:~/g/tacman/stenope-skeleton$ bin/console --version
Symfony 6.3.5 (env: dev, debug: true) #StandWithUkraine https://sf.to/ukraine
```

The assetmapper config is right out of the box:
```yaml
framework:
    asset_mapper:
        # The paths to make available to the asset mapper.
        paths:
            - assets/
```

Any suggestions on how to debug this?

**Comment by weaverryan** [comment, marked as answer]
Great debugger - makes my life easy :). 

I'm not sure why, but your `config/bundles.php` is missing the SwupBundle and so your AssetMapper is missing a "path" to the assets in SwupBundle (compare `bin/console debug:asset` before and after the fix). So, the fix is to add it there.

Why were you missing it? Previously, `symfony/ux-swup` had no bundle: it was just a package. We added a bundle class specifically to allow the AssetMapper path to be added. But if you had it previously installed, when you upgraded `symfony/ux-swup` to the version that DID have the bundle, my guess is that Flex didn't notice that, and so didn't add the bundle. Kind of a kink in the flow - so it was good question. Fortunately, iirc, this only affects ux-swup and ux-typed: most UX packages have always had a bundle class.

**Reply by tacman** [reply]
Thanks!  This was driving me crazy.  I hope this helps others who are migrating as well.  Of course, when I tried to reproduce the problem from a new project, everything worked as expected.
